lhotse/recipes/vcc2018.py

```diff
@@ -0,0 +1,329 @@
+    :param output_dir: Pathlike, the path where to write the manifests.
+    :return: a dict with keys "read" and "spontaneous".
+        Each hold another dict of {'recordings': ..., 'supervisions': ...}
+    """
+    corpus_dir = Path(corpus_dir)
+    assert corpus_dir.is_dir(), f"No such directory: {corpus_dir}"
+
+    converted_speech_dir = corpus_dir / "mnt/sysope/test_files/testVCC2/"
+    assert converted_speech_dir.is_dir(), f"No such directory: {converted_speech_dir}"
+
+    mos_dir = corpus_dir / "vcc2018_listening_test_scores"
+    assert mos_dir.is_dir(), f"No such directory: {mos_dir}"
+    mos_scores = mos_dir / "vcc2018_evaluation_mos.txt"
+    assert mos_scores.is_file()
+    sim_scores = mos_dir / "vcc2018_evaluation_sim.txt"
+    sim_scores.is_file()
+
+    trn_dir = corpus_dir / "scripts"
+    assert trn_dir.is_dir(), f"No such directory: {trn_dir}"
+
+    reference_speech_dir = corpus_dir / "vcc2018_reference"
+    assert reference_speech_dir.is_dir(), f"No such directory: {reference_speech_dir}"
+
+    logging.info(
+        f"Collecting reference target recordings for the VCC2018 challange from {reference_speech_dir}"
+    )
+
+    # TODO
+    # def from_file(p: Path, prefix):
+    #     # Match the format of converted recordings  B00_VCC2TF1_VCC2SF1_30001_HUB.wav
+    #     return Recording.from_file(
+    #         p, recording_id=f"reference_reference_{p.parent.stem}_{p.stem}_reference"
+    #     )
+    #
+    # reference_target_recordings = RecordingSet.from_recordings(
+    #     tqdm(
+    #         map(from_file, reference_speech_dir.rglob("*.wav")),
+    #         desc="Collecting reference recordings",
+    #     )
+    # )
+
+    logging.info(
+        f"Collecting converted recordings submitted to the VCC2018 challange from {converted_speech_dir}"
+    )
+    converted_recordings = RecordingSet.from_dir(
+        converted_speech_dir, "*.wav", num_jobs=num_jobs
+    )
+
+    # # TODO distinguish based on prefix
+    # # recordings = converted_recordings + reference_target_recordings
+    # recordings = reference_target_recordings
+    recordings = converted_recordings
+
+    id2trn = {}
+    for trnp in trn_dir.glob("*.txt"):
+        with open(trnp, "rt") as r:
+            id2trn[trnp.stem] = r.read().strip()
+
+    supervisions = prepare_mos_supervisions(mos_scores, recordings, id2trn)
+    validate_recordings_and_supervisions(recordings, supervisions)
+
+    if output_dir is not None:
+        output_dir = Path(output_dir)
+        output_dir.mkdir(parents=True, exist_ok=True)
+        recordings.to_jsonl(output_dir / "recordings.jsonl.gz")
+        supervisions.to_jsonl(output_dir / "supervisions.jsonl.gz")
+
+    return {"recordings": recordings, "supervisions": supervisions}
+
+
+def prepare_mos_supervisions(
+    mos_results_path, recordings: RecordingSet, id2trn: Dict[str, str]
+) -> SupervisionSet:
+    # TODO very slow -> make it faster it takes ~8min 170it/s
```

Comment:
This may be because you are manipulating a Pandas dataframe. Pandas does a lot of book-keeping under the hood which makes it good for complicated operations, but here I think it may be better to just use something like a list of namedtuples, and then use `groupby` to group them.